sdv/sampler.py

```diff
@@ -48,6 +63,15 @@ def transform_synthesized_rows(self, synthesized_rows, table_name, num_rows):
         meta = self.dn.tables[table_name].meta
         orig_meta = self._get_table_meta(self.dn.meta, table_name)
         primary_key = meta.get('primary_key')
+        categorical_fields = [
+            field['name'] for field in orig_meta['fields']
+            if field['type'] == 'categorical'
+        ]
+
+        if categorical_fields:
+            for field in categorical_fields:
+                if ((synthesized[field] < 0) | (synthesized[field] > 1)).any():
```

Comment:
I'd capture the `synthesized[field]` in a variable to avoid having to access it every time we use it.